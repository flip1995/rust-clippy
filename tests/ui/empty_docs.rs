#![allow(unused)]
#![warn(clippy::empty_docs)]

/// this is a struct
struct Bananas {
    /// count
    count: usize,
}

///
enum Warn {
    ///
    A,
    ///
    B,
}

enum WarnForB {
    /// it's ok
    A,
    ///
    B,
}

#[doc = ""]
#[doc = ""]
fn warn_about_this() {}

#[doc = "a fine function"]
fn this_is_fine() {}

fn warn_about_this_as_well() {
    //!
}

fn this_is_ok() {
    //!
    //! inside the function
}

fn warn() {
    /*! inside the function */
}
