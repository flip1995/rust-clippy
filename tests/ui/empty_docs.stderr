error: empty doc comment
  --> tests/ui/empty_docs.rs:10:1
   |
LL | ///
   | ^^^
   |
   = help: consider removing or filling it
   = note: `-D clippy::empty-docs` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::empty_docs)]`

error: empty doc comment
  --> tests/ui/empty_docs.rs:31:1
   |
LL | #[doc = ""]
   | ^^^^^^^^^^^
   |
   = help: consider removing or filling it

error: empty doc comment
  --> tests/ui/empty_docs.rs:34:1
   |
LL | / #[doc = ""]
LL | | #[doc = ""]
   | |___________^
   |
   = help: consider removing or filling it

error: empty doc comment
  --> tests/ui/empty_docs.rs:42:5
   |
LL |     //!
   |     ^^^
   |
   = help: consider removing or filling it

error: empty doc comment
  --> tests/ui/empty_docs.rs:56:5
   |
LL |     /*! */
   |     ^^^^^^
   |
   = help: consider removing or filling it

error: aborting due to 5 previous errors

